libs/model/src/main/kotlin/de/cyface/model/RequestMetaData.kt

```diff
@@ -19,148 +19,301 @@
+        val videoCount: Int,
+        val filesSize: Long,
+    ) : Serializable {
+        init {
+            require(logCount >= 0) { "Invalid logCount: $logCount" }
+            require(imageCount >= 0) { "Invalid imageCount: $imageCount" }
+            require(videoCount >= 0) { "Invalid videoCount: $videoCount" }
+            require(filesSize >= 0) { "Invalid filesSize: $filesSize" }
+        }
 
+        companion object {
+            /**
+             * Used to serialize objects of this class. Only change this value if this classes attribute set changes.
+             */
+            @Suppress("ConstPropertyName")
+            private const val serialVersionUID = 1L
+        }
+    }
+
+    companion object {
         /**
-         * The default char set to use for encoding and decoding strings transmitted as metadata.
+         * Used to serialize objects of this class. Only change this value if this classes attribute set changes.
          */
-        private const val DEFAULT_CHARSET = "UTF-8"
+        @Suppress("ConstPropertyName")
+        private const val serialVersionUID = 2L
 
         /**
-         * Maximum size of a metadata field, with plenty space for future development. This prevents attackers from putting
+         * Maximum size of a metadata field, with plenty space for future development. Prevents attackers from putting
          * arbitrary long data into these fields.
          */
         const val MAX_GENERIC_METADATA_FIELD_LENGTH = 30
+    }
 
-        /**
-         * The maximum length of the measurement identifier in characters (this is the amount of characters of
-         * {@value Long#MAX_VALUE}).
-         */
-        private const val MAX_MEASUREMENT_ID_LENGTH = 20
+    override fun equals(other: Any?): Boolean {
+        if (this === other) return true
+        if (javaClass != other?.javaClass) return false
 
-        /**
-         * The minimum length of a track stored with a measurement.
-         */
-        private const val MINIMUM_TRACK_LENGTH = 0.0
+        other as RequestMetaData<*>
 
-        /**
-         * The minimum valid amount of locations stored inside a measurement.
-         */
-        private const val MINIMUM_LOCATION_COUNT = 0L
+        if (identifier != other.identifier) return false
+        if (deviceMetaData != other.deviceMetaData) return false
+        if (applicationMetaData != other.applicationMetaData) return false
+        if (measurementMetaData != other.measurementMetaData) return false
+        if (attachmentMetaData != other.attachmentMetaData) return false
 
-        /**
-         * The current version of the transferred file. This is always specified by the first two bytes of the file
-         * transferred and helps compatible APIs to process data from different client versions.
-         */
-        const val CURRENT_TRANSFER_FILE_FORMAT_VERSION = 3
+        return true
+    }
+
+    override fun hashCode(): Int {
+        return identifier.hashCode()
     }
```

Comment:
Added as 2 RequestMetaData instances should only be different when the identifier is different, either deviceMeasurementId for a measurement upload or the deviceAttachementId for the attachment upload.